src/math/src/tests/math_test.cairo

```diff
@@ -170,3 +170,49 @@ fn shl_should_not_overflow() {
     assert(BitShift::shl(pow::<u128>(2, 127), 1) == 0, 'invalid result');
     assert(BitShift::shl(pow::<u256>(2, 255), 1) == 0, 'invalid result');
 }
+
+#[test]
+#[available_gas(2000000)]
+fn test_rotl_min() {
+    assert(BitRotate::rotl(pow::<u8>(2, 7) + 1, 1) == 3, 'invalid result');
+    assert(BitRotate::rotl(pow::<u16>(2, 15) + 1, 1) == 3, 'invalid result');
+    assert(BitRotate::rotl(pow::<u32>(2, 31) + 1, 1) == 3, 'invalid result');
+    assert(BitRotate::rotl(pow::<u64>(2, 63) + 1, 1) == 3, 'invalid result');
+    assert(BitRotate::rotl(pow::<u128>(2, 127) + 1, 1) == 3, 'invalid result');
+//  TODO(sveamarcus): missing implementation
```

Comment:
Still need todo something here?

Reply:
Not sure what to do about the 256 bit implementation, see the comment in lib.cairo under math. There is a missing `u512_as_non_zero` call in Cairo. Maybe I should raise an issue there? Alternatively, using bitwise `AND` plus shifts could provide a different solution for this specific case.

Reply:
Added a different traditional implementation for `rotate_left` `u256`